test/src/main/java/org/apache/accumulo/test/fate/zookeeper/FateIT.java

```diff
@@ -0,0 +1,193 @@
+    zk.mkdirs(ZK_ROOT + Constants.ZTABLE_LOCKS);
+    zk.mkdirs(ZK_ROOT + Constants.ZNAMESPACES + "/" + NS.canonical());
+    zk.mkdirs(ZK_ROOT + Constants.ZTABLE_STATE + "/" + TID.canonical());
+    zk.mkdirs(ZK_ROOT + Constants.ZTABLES + "/" + TID.canonical());
+
+    ZooStore<Manager> zooStore = new ZooStore<Manager>(ZK_ROOT + Constants.ZFATE, zk);
+    final AgeOffStore<Manager> store = new AgeOffStore<Manager>(zooStore, 1000 * 60 * 60 * 8);
+
+    Manager manager = createMock(Manager.class);
+    ServerContext sctx = createMock(ServerContext.class);
+    expect(manager.getContext()).andReturn(sctx).anyTimes();
+    expect(sctx.getZooKeeperRoot()).andReturn(ZK_ROOT).anyTimes();
+    expect(sctx.getZooReaderWriter()).andReturn(zk).anyTimes();
+    replay(manager, sctx);
+
+    Fate<Manager> fate = new Fate<Manager>(manager, store, TraceRepo::toLogString);
+    try {
+      ConfigurationCopy config = new ConfigurationCopy();
+      config.set(Property.GENERAL_SIMPLETIMER_THREADPOOL_SIZE, "2");
+      config.set(Property.MANAGER_FATE_THREADPOOL_SIZE, "1");
+      fate.startTransactionRunners(config);
+
+      // Wait for the transaction runner to be scheduled.
+      UtilWaitThread.sleep(3000);
+
+      callStarted = new CountDownLatch(1);
+      finishCall = new CountDownLatch(1);
+
+      long txid = fate.startTransaction();
+      assertEquals(TStatus.NEW, getTxStatus(zk, txid));
+      fate.seedTransaction(txid, new TestOperation(NS, TID), true);
+      assertEquals(TStatus.SUBMITTED, getTxStatus(zk, txid));
+      // wait for call() to be called
+      callStarted.await();
+      assertEquals(TStatus.IN_PROGRESS, getTxStatus(zk, txid));
+      // tell the op to exit the method
+      finishCall.countDown();
+      // Check that it transitions to SUCCESSFUL
+      TStatus s = getTxStatus(zk, txid);
+      while (!s.equals(TStatus.SUCCESSFUL)) {
+        s = getTxStatus(zk, txid);
+      }
+      // Check that it gets removed
+      boolean errorSeen = false;
+      while (!errorSeen) {
+        try {
+          s = getTxStatus(zk, txid);
+        } catch (KeeperException e) {
+          if (e.code() == KeeperException.Code.NONODE) {
+            errorSeen = true;
+          } else {
+            fail("Unexpected error thrown: " + e.getMessage());
+          }
+        }
+      }
+
+    } finally {
+      fate.shutdown();
+    }
+  }
+
+  public static void inCall() throws InterruptedException {
+    // signal that call started
+    callStarted.countDown();
+    // wait for the signal to exit the method
+    finishCall.await();
+  }
+
+  /*
+   * Get the status of the TX from ZK directly. Unable to call ZooStore.getStatus because this test
+   * thread does not have the reservation (the FaTE thread does)
+   */
+  private static TStatus getTxStatus(ZooReaderWriter zrw, long txid)
+      throws KeeperException, InterruptedException {
+    String txdir = String.format("%s/tx_%016x", ZK_ROOT + Constants.ZFATE, txid);
```

Comment:
This is already using a format string. So, you don't need to do the concatenation as well:
```suggestion
    String txdir = String.format("%s%s/tx_%016x", ZK_ROOT, Constants.ZFATE, txid);
```

Reply:
Addressed in 46a503ace773ab8f3c1f35a91691c78014e43f9b